src/core/playground/accountAuthenticator/accountAuthenticator.ts

```diff
@@ -0,0 +1,21 @@
+import type { Deserializer } from "../../../bcs";
+import { LegacyAccountAuthenticatorEd25519 } from "./legacyEd25519";
+import { LegacyAccountAuthenticatorMultiEd25519 } from "./legacyMultiEd25519";
+import { AccountAuthenticatorSingleKey } from "./singleKey";
+import { AccountAuthenticatorVariant } from "./variant";
+
+export class AccountAuthenticator {
```

Comment:
Hm, so this is not an abstract class anymore?

Reply:
It's kinda hard to achieve this without cyclical dependency :(

Reply:
only solution I could find is to have everything in the same file, which is ok but not really scalable

Reply:
I thought "what's requiring other authenticators to extend from a base AccountAuthenticator?" and removed the "extends" for now